File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
  stack<int> stack_q;
  stack<int> buffer_q;

 public:
  void push(int x) { stack_q.push(x); }
  int pop() {
    if (buffer_q.empty()) {
      while (!stack_q.empty()) {
        buffer_q.push(stack_q.top());
        stack_q.pop();
      }
    }
    int temp = buffer_q.top();
    buffer_q.pop();
    return temp;
  }
  int peek() {
    int temp = 0;
    if (!buffer_q.empty()) {
      temp = buffer_q.top();
    } else {
      while (!stack_q.empty()) {
        buffer_q.push(stack_q.top());
        stack_q.pop();
      }
      temp = buffer_q.top();
    }
    return temp;
  }
  bool empty() {
    if (buffer_q.empty() && stack_q.empty())
      return true;
    else
      return false;
  }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: Python/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.s1 = []
        self.s2 = []

    def push(self, x: int) -> None:
        while self.s1:
            self.s2.append(self.s1.pop())
        self.s1.append(x)
        while self.s2:
            self.s1.append(self.s2.pop())

    def pop(self) -> int:
        return self.s1.pop()

    def peek(self) -> int:
        return self.s1[-1]

    def empty(self) -> bool:
        return not self.s1


# Your MyQueue object will be instantiated and called as such:
obj = MyQueue()
obj.push(1)
obj.push(2)
obj.peek()
obj.pop()
obj.empty()
